internal/pkg/ctxattr/ctxattr.go

```diff
@@ -0,0 +1,90 @@
+package ctxattr
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.uber.org/zap"
+)
+
+type ctxKey string
+
+const (
+	contextAttributes = ctxKey("contextAttributes")
+	contextZapFields  = ctxKey("contextZapFields")
+)
+
+func ContextWith(ctx context.Context, attributes ...attribute.KeyValue) context.Context {
+	// Add attributes to context.
+	var newAttributes []attribute.KeyValue
+	value := ctx.Value(contextAttributes)
+	if value != nil {
+		newAttributes = value.(*attribute.Set).ToSlice()
+		newAttributes = append(newAttributes, attributes...)
+	} else {
+		newAttributes = attributes
+	}
+
+	set := attribute.NewSet(newAttributes...)
+	ctx = context.WithValue(ctx, contextAttributes, &set)
+
+	// Add attributes as zap fields for logger.
+	zapFields := make([]zap.Field, set.Len())
+	for i, keyValue := range set.ToSlice() {
+		zapFields[i] = convertAttributeToZapField(keyValue)
+	}
+
+	return context.WithValue(ctx, contextZapFields, zapFields)
+}
+
+func Attributes(ctx context.Context) *attribute.Set {
+	if ctx == nil {
+		return attribute.EmptySet()
+	}
+
+	value := ctx.Value(contextAttributes)
+	if value != nil {
+		return value.(*attribute.Set)
+	}
+
+	return attribute.EmptySet()
+}
+
+func ZapFields(ctx context.Context) []zap.Field {
+	if ctx == nil {
+		return []zap.Field{}
+	}
+
+	value := ctx.Value(contextZapFields)
+	if value != nil {
+		return value.([]zap.Field)
+	}
```

Comment:
Type assertion can be written in two ways:
- `value := interface.(myType)` - if the interface is not the `myType`, a panic occurs
- `value, ok := interface.(myType)` - if the interface is not the `myType`, the value is empty and `ok=false`
  - this is safer
  
---------------------
  
So I prefer:
```
if v, ok := ctx.Value(contextZapFields).([]zap.Field); ok {
    return v
}
```

But in your code, in this particular case, both ways are safe.

Reply:
![image](https://github.com/keboola/keboola-as-code/assets/19371734/aff66e1a-56be-484d-91aa-8f489fa70ee5)